src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11769,54 +11774,53 @@ internal ExpressionSyntax ParseConsequenceSyntax()
                 _ => throw ExceptionUtilities.Unreachable(),
             };
 
-            while (true)
+            while (tryExpandExpression(expr) is ExpressionSyntax expandedExpression)
+                expr = expandedExpression;
+
+            return expr;
+
+            ExpressionSyntax tryExpandExpression(ExpressionSyntax expr)
             {
-                // Nullable suppression operators should only be consumed by a conditional access
-                // if there are further conditional operations performed after the suppression
-                if (isOptionalExclamationsFollowedByConditionalOperation())
-                {
-                    while (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
-                        expr = _syntaxFactory.PostfixUnaryExpression(SyntaxKind.SuppressNullableWarningExpression, expr, EatToken());
-                }
+                using var beforeSuppressionsResetPoint = GetDisposableResetPoint(resetOnDispose: false);
+
+                // consume any '!'s from the preceding conditional operation.
```

Comment:
can you explain wym by "conditional operation" here?

Reply:
*conditional operation* is one of the expressions we parsed out from a previous call to `tryExpandExpression`, such `.b` or `.b.c` in `a?.b.c`. It corresponds to [*dependent-access*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1177-null-conditional-member-access) in the standard. Possibly, we could change the term now to match the standard.

The idea is that while `a?.b!` should parse as "a suppression containing a conditional access", `a?.b!.c` should parse such that `.b!.c` is the RHS of the conditional access.

Reply:
this is a great comment that would be good to put into the PR :)